### PR TITLE
Fix Programmatic_Subscribe_DynamicAddNewStreamProvider_WhenConsuming

### PR DESCRIPTION
**Pull request by xiazen**
tests failed because I'm not aware the power of Stateless worker...

1. remove StatelessWorker on the consumer grain, since tests rely on the consumer grain to have state, using StatelessWorker is unnecessary 
2. added logging to producer grain , for easier debugging in the future 
3. added the failing test back to BVT and Functionals

### CODE REVIEW

File: test/Tester/StreamingTests/ProgrammaticSubscribeTests/ProgrammaticSubscribeTestsWithDynamicProviderConfiguration.cs [hunk 1]
```diff
@@ -104,19 +104,27 @@ public ProgrammaticSubscribeTestsWithDynamicProviderConfiguration(ITestOutputHel
...
+                tasks.Add(consumer.SetupOnSubscriptionChangActionForProvider(StreamProviderName2));
+            }
+            await Task.WhenAll(tasks);
+            tasks.Clear();
             //set up producer2 to produce to stream2
             var producer2 = this.GrainFactory.GetGrain<ITypedProducerGrainProducingString>(Guid.NewGuid());
             await producer2.BecomeProducer(streamId2.Guid, streamId2.Namespace, streamId2.ProviderName);
 
-            await producer2.StartPeriodicProducing(TimeSpan.FromSeconds(1));
+            await producer2.StartPeriodicProducing();
```
Comment (xiazen) [inline test/Tester/StreamingTests/ProgrammaticSubscribeTests/ProgrammaticSubscribeTestsWithDynamicProviderConfiguration.cs]: changed the frequency to default every 30 miliseconds

File: test/TestGrainInterfaces/ProgramaticStreamSubscribe/IStateless_ConsumerGrain.cs [hunk 2]
```diff
@@ -14,6 +14,7 @@ public interface IStateless_ConsumerGrain: IGrainWithGuidKey
         Task StopConsuming();
         Task<int> GetCountOfOnAddFuncCalled();
         Task<int> GetNumberConsumed();
+        Task SetupOnSubscriptionChangActionForProvider(string providerName);
```
Comment (jason-bragg) [inline test/TestGrainInterfaces/ProgramaticStreamSubscribe/IStateless_ConsumerGrain.cs]: Typo - Chang => Change

File: test/TestGrains/ProgrammaticSubscribe/Stateless_ConsumerGrain.cs [hunk 3]
```diff
@@ -9,10 +9,8 @@
 
 namespace UnitTests.Grains
 {
-    [StatelessWorker(MaxLocalWorkers)]
```
Comment (jason-bragg) [inline test/TestGrains/ProgrammaticSubscribe/Stateless_ConsumerGrain.cs]: grain name is stateless consumer, but it's no longer stateless?

**Comment by jason-bragg** [comment]
minor comments, LG2M

**Comment by xiazen** [comment]
feedback addressed

**Pull request merged**